File: hackerrank/easy/0001-day-of-the-programmer/js/solution.mjs
// solution.mjs

export function dayOfProgrammer(year) {
  if (year === 1918) return `26.09.${year}`;
  if (year < 1918) {
    return year % 4 === 0 ? `12.09.${year}` : `13.09.${year}`;
  }
  if (year % 400 === 0 || (year % 4 === 0 && year % 100 !== 0)) {
    return `12.09.${year}`;
  }
  return `13.09.${year}`;
}

// --- demo run ---
if (import.meta.url === `file://${process.argv[1]}`) {
  console.log(dayOfProgrammer(2017)); // 13.09.2017
}

File: hackerrank/easy/0001-day-of-the-programmer/python/solution.py
// solution.py

def day_of_programmer(year: int) -> str:
    if year == 1918:
        return f"26.09.{year}"
    elif year < 1918:
        return f"{'12.09' if year % 4 == 0 else '13.09'}.{year}"
    else:
        if year % 400 == 0 or (year % 4 == 0 and year % 100 != 0):
            return f"12.09.{year}"
        else:
            return f"13.09.{year}"


# --- demo run ---
if __name__ == "__main__":
    print(day_of_programmer(2017))  # 13.09.2017